src/plugin/context/mod.rs

```diff
@@ -327,19 +360,42 @@ impl RapierContext {
                         &mut self.ccd_solver,
                         None,
                         hooks,
-                        events,
+                        event_handler,
                     );
                     executed_steps += 1;
                 }
             }
         }
+        if let Some(mut event_queue) = event_queue {
+            // SAFETY: event_queue and its inner locksare only accessed from
+            // within `self.pipeline.step` called above, so we can unwrap here safely.
```

Comment:
The `SAFETY` statement is typically for unsafe code. Unwrapping isn’t unsafe.
```suggestion
            // NOTE: the event_queue and its inner locks are only accessed from
            // within `self.pipeline.step` called above, so we can unwrap here safely.
```